### PR TITLE
Add extensions listing page and convert APIs to TS

### PR DESCRIPTION
**Pull request by mythmon**
Ref #535

### CODE REVIEW

**Comment by mythmon** [comment]
Oh no. That failing test is an intermittent. I also realized I forgot to add any new tests for the new page. Oops!

File: content/components/pages/ExtensionsPage.tsx [hunk 1]
```diff
@@ -0,0 +1,156 @@
...
+  Icon,
+  Nav,
+  Navbar,
+  Pagination,
+  Panel,
+  Row,
+  Tag,
+} from "rsuite";
+
+import AsyncHookView from "../common/AsyncHookView";
```
Comment (rehandalal) [inline content/components/pages/ExtensionsPage.tsx]: fix relative path

File: content/components/App.js [hunk 2]
```diff
@@ -16,13 +16,16 @@ import RecipesPage from "devtools/components/pages/RecipesPage";
...
+import NotFoundPage from "./pages/NotFoundPage";
+
 // RecipesPage is not lazy since it is the default route.
 const RecipeFormPage = React.lazy(() =>
   import("devtools/components/pages/RecipeFormPage"),
 );
 const RecipeDetailsPage = React.lazy(() =>
   import("devtools/components/pages/RecipeDetailsPage"),
 );
+const ExtensionsPage = React.lazy(() => import("./pages/ExtensionsPage"));
```
Comment (rehandalal) [inline content/components/App.js]: fix relative path

File: content/components/App.js [hunk 3]
```diff
@@ -16,13 +16,16 @@ import RecipesPage from "devtools/components/pages/RecipesPage";
 import { EnvironmentProvider } from "devtools/contexts/environment";
 import { useHistoryRecorder } from "devtools/hooks/urls";
 
+import NotFoundPage from "./pages/NotFoundPage";
```
Comment (rehandalal) [inline content/components/App.js]: fix relative path

File: content/components/pages/RecipeDetailsPage.tsx [hunk 4]
```diff
@@ -10,9 +10,10 @@ import {
 import { ExperimenterDetailsProvider } from "devtools/contexts/experimenterDetails";
 import { RecipeDetailsProvider } from "devtools/contexts/recipeDetails";
 
+import NotFoundPage from "./NotFoundPage";
```
Comment (rehandalal) [inline content/components/pages/RecipeDetailsPage.tsx]: fix relative path

File: content/components/pages/RecipeFormPage.tsx [hunk 5]
```diff
@@ -14,15 +14,29 @@ import {
   RecipeDetailsProvider,
 } from "devtools/contexts/recipeDetails";
 
+import NotFoundPage from "./NotFoundPage";
```
Comment (rehandalal) [inline content/components/pages/RecipeFormPage.tsx]: fix relative path

File: content/tests/RecipeForm.test.js [hunk 6]
```diff
@@ -10,6 +10,7 @@ import {
 import ExperimenterAPI from "devtools/utils/experimenterApi";
 import NormandyAPI from "devtools/utils/normandyApi";
 
+import { experimenterResponseFactory } from "./factories/experiments";
```
Comment (rehandalal) [inline content/tests/RecipeForm.test.js]: fix relative path

File: content/utils/api.ts [hunk 7]
```diff
@@ -1,37 +1,46 @@
+import { Environment } from "devtools/contexts/environment";
 import { SECOND } from "devtools/utils/timeConstants";
 
+import { has } from "./helpers";
```
Comment (rehandalal) [inline content/utils/api.ts]: fix relative path

**Review by rehandalal — DISMISSED** [review]
nits

**Review by rehandalal — APPROVED** [review]
this lgtm but for failing tests

**Comment by mythmon** [comment]
bors r= @rehandalal

**Pull request merged**